packages/service-library/src/servicelib/fastapi/long_running/_client.py

```diff
@@ -0,0 +1,94 @@
+from functools import cached_property
+from typing import Any, Optional
+
+from fastapi import FastAPI, status
+from httpx import AsyncClient, Timeout
+from pydantic import AnyHttpUrl, PositiveFloat
+
+from ._errors import TaskClientResultErrorError
+from ._models import TaskId, TaskResult, TaskStatus
+
+
+class Client:
+    def __init__(
+        self,
+        *,
+        prefix: str,
+        timeout: PositiveFloat,
+        status_poll_interval: PositiveFloat,
+    ) -> None:
+        self._timeout = Timeout(timeout)
+        self._prefix = prefix
+        self._status_poll_interval = status_poll_interval
+
+    @cached_property
+    def status_poll_interval(self) -> PositiveFloat:
+        return self._status_poll_interval
+
+    def _get_url(self, base_url: AnyHttpUrl, path: str) -> str:
+        return f"{base_url}{self._prefix}{path}"
+
+    async def get_task_status(
+        self, async_client: AsyncClient, base_url: AnyHttpUrl, task_id: TaskId
+    ) -> TaskStatus:
+        result = await async_client.get(
+            self._get_url(base_url, f"/task/{task_id}"), timeout=self._timeout
+        )
+        assert result.status_code == status.HTTP_200_OK  # nosec
+        return TaskStatus.parse_obj(result.json())
+
+    async def get_task_result(
+        self, async_client: AsyncClient, base_url: AnyHttpUrl, task_id: TaskId
+    ) -> Optional[Any]:
+        result = await async_client.get(
+            self._get_url(base_url, f"/task/{task_id}/result"), timeout=self._timeout
```

Comment:
I manage to unify a bit the error handling and it should feel better now.

Regarding task.result() please see how the result is recovered. There are extensive tests there. In short:
- first che task is checked that it is not cancelled (an error will be raised in such a case)
- if the task finished with an error, it will be encoded and returned `TaskResult`
- if the task has a result it will be encoded and returned in the `TaskResult`
- if the task is not found a 404 is raised